solidity/dashboard/src/components/AddTopUpModal.jsx

```diff
@@ -1,39 +1,209 @@
+  authorizerAddress,
+  beneficiary,
+  operatorAddress,
+  currentAmount,
+  availableAmount,
+  onSubmit,
+  onCancel,
 }) => {
   return (
     <>
       <h3 className="mb-1">
         Enter an amount of KEEP to add to this existing delegation.
       </h3>
-      <span className="text-big text-grey-70">Delegation Details</span>
-      <DelegationAddress address={authorizerAddress} label="authorizer" />
-      <DelegationAddress address={operatorAddress} label="operator" />
-      <DelegationAddress address={beneficiary} label="beneficiary" />
-      <AmountForm {...formProps} />
+      <h4 className="text-grey-70 mb-1">Delegation Details</h4>
+      <DelegationDetails
+        authorizerAddress={authorizerAddress}
+        beneficiary={beneficiary}
+        operatorAddress={operatorAddress}
+      />
+      <div className="flex row center mt-2 mb-2">
+        <div className="flex-1">
+          <Tag text="Current" IconComponent={Icons.KeepToken} />
+        </div>
+        <TokenAmount amount={currentAmount} />
+      </div>
+      <AmountForm
+        onCancel={onCancel}
+        submitBtnText="add keep"
+        availableAmount={availableAmount}
+        currentAmount={currentAmount}
+        onBtnClick={onSubmit}
+      />
+    </>
+  )
+}
+
+const Step2Content = ({
+  authorizerAddress,
+  beneficiary,
+  operatorAddress,
+  amountToAdd,
+  newAmount,
+}) => {
+  return (
+    <>
+      <TokenAmount
+        amount={KEEP.fromTokenUnit(amountToAdd).toString()}
+        withIcon
+        icon={Icons.Plus}
+        iconProps={{ width: 24, height: 24, className: "plus-icon" }}
+      />
+      <h4 className="text-grey-70 mb-1">
+        New delegation balance: {KEEP.displayAmountWithSymbol(newAmount)}
+      </h4>
+      <DelegationDetails
+        authorizerAddress={authorizerAddress}
+        beneficiary={beneficiary}
+        operatorAddress={operatorAddress}
+      />
     </>
   )
 }
 
+const Step2 = withConfirmationModal(Step2Content)
+
 const DelegationAddress = ({ address, label }) => {
   return (
-    <div className="flex row center">
-      <div className="text-label text-small text-grey-70 flex-1">{label}</div>
-      <div className="flex-2 self-start">
-        <AddressShortcut
-          address={address}
-          classNames="text-label text-grey-50"
-        />
-      </div>
+    <div className="flex row center text-grey-70">
+      <h5 className="flex-1">{label}</h5>
+      <AddressShortcut address={address} classNames="h6 text-grey-70" />
     </div>
   )
 }
 
+const DelegationDetails = ({
+  authorizerAddress,
+  beneficiary,
+  operatorAddress,
+}) => (
+  <>
+    <DelegationAddress address={authorizerAddress} label="authorizer" />
+    <DelegationAddress address={operatorAddress} label="operator" />
+    <DelegationAddress address={beneficiary} label="beneficiary" />
+  </>
+)
+
+const TopUpInitiatedConfirmationModal = ({
+  // In token unit.
+  addedAmount,
+  // In the smallest token unit.
+  currentAmount,
```

Comment:
Just wondering - wouldn't it be better to keep the amount props in the same unit? We could change it when passing the values to the props instead of doing it inside component. I think it would be better for readability.